Azure/sonic-buildimage
# Tests for bgpcfgd templates

**- Why I did it**
1. Implemented tests for bgpd templates. Tests are checking for correct rendering of the templates and also they check that every ipv6 neighbor has 'set ipv6 prefer-global' configuration
2.  Extract modules from bgpcfgd. This was done to write the test, but this work is not completed. I'm going to complete this work in the next PR.

**- How I did it**
I used pytest framework to write the tests.

**- How to verify it**
```
$ cd src/sonic-bgpcfgd
$ python setup.py test
```
It should give you following output:
```
running pytest
running egg_info
writing requirements to sonic_bgpcfgd.egg-info/requires.txt
writing sonic_bgpcfgd.egg-info/PKG-INFO
writing top-level names to sonic_bgpcfgd.egg-info/top_level.txt
writing dependency_links to sonic_bgpcfgd.egg-info/dependency_links.txt
reading manifest file 'sonic_bgpcfgd.egg-info/SOURCES.txt'
writing manifest file 'sonic_bgpcfgd.egg-info/SOURCES.txt'
running build_ext
============================= test session starts ==============================
platform linux2 -- Python 2.7.12, pytest-4.6.9, py-1.8.1, pluggy-0.13.1 -- /usr/bin/python
cachedir: .pytest_cache
rootdir: /home/ps/work/sonic-buildimage/src/sonic-bgpcfgd, inifile: setup.cfg
plugins: flaky-3.6.1
collecting ... collected 10 items

tests/test_ipv6_nexthop_global.py::test_v6_next_hop_global PASSED        [ 10%]
tests/test_templates.py::test_general_policies PASSED                    [ 20%]
tests/test_templates.py::test_general_pg PASSED                          [ 30%]
tests/test_templates.py::test_general_instance PASSED                    [ 40%]
tests/test_templates.py::test_dynamic_policies PASSED                    [ 50%]
tests/test_templates.py::test_dynamic_pg PASSED                          [ 60%]
tests/test_templates.py::test_dynamic_instance PASSED                    [ 70%]
tests/test_templates.py::test_monitors_policies PASSED                   [ 80%]
tests/test_templates.py::test_monitors_pg PASSED                         [ 90%]
tests/test_templates.py::test_monitors_instance PASSED                   [100%]

========================== 10 passed in 0.27 seconds ===========================

```

**- Description for the changelog**



**- A picture of a cute animal (not mandatory but encouraged)**
